### PR TITLE
Modern_diag_manager- min/max

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
- Implement the min and max reduction methods into diag manager
- modify tests to test the new diag manager

Fixes # (issue)

**How Has This Been Tested?**
CI, including updated tests

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: diag_manager/include/fms_diag_reduction_methods.inc [hunk 1]
```diff
@@ -54,4 +54,107 @@ subroutine DO_TIME_NONE_ (data_out, data_in, mask, bounds_in, bounds_out, missin
...
+  ie_in = bounds_in%get_imax()
+  js_in = bounds_in%get_jmin()
+  je_in = bounds_in%get_jmax()
+  ks_in = bounds_in%get_kmin()
+  ke_in = bounds_in%get_kmax()
+
+  do i = 0, ie_out - is_out
+    do j = 0, je_out - js_out
+      do k = 0, ke_out - ks_out
+        do l = 0, size(data_out, 4) - 1
```
Comment (thomas-robinson) [inline diag_manager/include/fms_diag_reduction_methods.inc:94]: Is this backwards?

**Review by thomas-robinson — CHANGES_REQUESTED** [review]
I think your loops in fms_diag_reduction_methods.inc need to be reordered

File: diag_manager/include/fms_diag_reduction_methods.inc (hunk 1 above)
Comment (uramirez8707) [reply diag_manager/include/fms_diag_reduction_methods.inc:94]: should it be l, k, j, i?

Comment (uramirez8707) [reply diag_manager/include/fms_diag_reduction_methods.inc]: https://github.com/NOAA-GFDL/FMS/pull/1367/commits/4193a084aba6e5b6a2535bddef0928b49db0ddb2

File: diag_manager/include/fms_diag_reduction_methods.inc [hunk 2]
```diff
@@ -140,21 +166,39 @@ subroutine DO_TIME_MAX_ (data_out, data_in, mask, bounds_in, bounds_out, missing
...
+              endif
+            else
+              data_out(is_out + i, js_out + j, ks_out + k, l + 1, 1) = missing_value
+            endif
+          enddo
+        enddo
+      enddo
+    enddo
+  else
+    do l = 0, size(data_out, 4) - 1
```
Comment (thomas-robinson) [inline diag_manager/include/fms_diag_reduction_methods.inc:190]: Can this be a `do concurrent`?

File: diag_manager/include/fms_diag_reduction_methods.inc [hunk 3]
```diff
@@ -88,31 +95,50 @@ subroutine DO_TIME_MIN_ (data_out, data_in, mask, bounds_in, bounds_out, missing
...
```
Comment (thomas-robinson) [inline diag_manager/include/fms_diag_reduction_methods.inc:119]: Can this be a `do concurrent`?

**Review by thomas-robinson — COMMENTED** [review]
Can some of the loops be parallelized?

**Review by thomas-robinson — APPROVED** [review]

**Pull request merged**